lang/src/org/partiql/lang/eval/visitors/StaticTypeInferenceVisitorTransform.kt

```diff
@@ -269,64 +271,36 @@ internal class StaticTypeInferenceVisitorTransform(
             return node.withStaticType(foundType)
         }
 
-        private fun transformNAry(
```

Comment:
The diff for this PR is pretty hard to read because there's both refactoring of logic within functions and rearrangement of function order. One way to make the diff somewhat easier for reviewers to review could be to have two distinct commits: 1. performs logic changes/inner function refactoring. 2. rearranges the functions